frontend/packages/console-plugin-sdk/src/typings/features.ts

```diff
@@ -3,14 +3,17 @@ import { K8sKind } from '@console/internal/module/k8s';
 import { Extension } from './extension';
 
 namespace ExtensionProperties {
-  export interface ModelFeatureFlag {
-    /** If a CRD for this model exists, the feature will be enabled. */
-    model: K8sKind;
+  interface FeatureFlag {
     /** The name of the feature flag. */
     flag: string;
   }
 
-  export interface ActionFeatureFlag {
+  export interface ModelFeatureFlag extends FeatureFlag {
+    /** If a CRD for this model exists, the feature will be enabled. */
+    model: K8sKind;
+  }
+
+  export interface ActionFeatureFlag extends FeatureFlag {
     /** Function used to detect the feature and set flag name/value via Redux action dispatch. */
     detect: (dispatch: Dispatch) => Promise<any>;
```

Comment:
Some plugins might need to support k8sWatch which does not return a promise.
```suggestion
    detect: (dispatch: Dispatch) => Promise<any> | void;
```